internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestPushJobWithHeartBeatSystemStore.java

```diff
@@ -0,0 +1,224 @@
+    String parentControllerUrls =
+        parentControllers.stream().map(VeniceControllerWrapper::getControllerUrl).collect(Collectors.joining(","));
+    String inputDirPath = "file:" + inputDir.getAbsolutePath();
+    String storeName = Utils.getUniqueString("store");
+    Properties props = TestWriteUtils.defaultVPJProps(parentControllerUrls, inputDirPath, storeName);
+    props.put(SEND_CONTROL_MESSAGES_DIRECTLY, true);
+    props.put(
+        VENICE_CHILD_CONTROLLER_PROP,
+        childDatacenters.get(0).getControllerConnectString() + ','
+            + childDatacenters.get(1).getControllerConnectString());
+
+    UpdateStoreQueryParams updateStoreParams =
+        new UpdateStoreQueryParams().setStorageQuotaInByte(Store.UNLIMITED_STORAGE_QUOTA)
+            .setPartitionCount(partitionCount)
+            .setIncrementalPushEnabled(true);
+
+    Schema recordSchema = TestWriteUtils.writeSimpleAvroFileWithUserSchema(inputDir, true, recordCount);
+    String keySchemaStr = recordSchema.getField(DEFAULT_KEY_FIELD_PROP).schema().toString();
+    String valueSchemaStr = recordSchema.getField(DEFAULT_VALUE_FIELD_PROP).schema().toString();
+
+    try {
+      createStoreForJob(clusterName, keySchemaStr, valueSchemaStr, props, updateStoreParams).close();
+      childDatacenters.get(0)
+          .getClusters()
+          .get(clusterName)
+          .useControllerClient(
+              dc0Client -> childDatacenters.get(1)
+                  .getClusters()
+                  .get(clusterName)
+                  .useControllerClient(
+                      dc1Client -> TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, true, () -> {
+                        // verify the update store command has taken effect before starting the push job.
+                        Assert.assertEquals(
+                            dc0Client.getStore(storeName).getStore().getStorageQuotaInByte(),
+                            Store.UNLIMITED_STORAGE_QUOTA);
+                        Assert.assertEquals(
+                            dc1Client.getStore(storeName).getStore().getStorageQuotaInByte(),
+                            Store.UNLIMITED_STORAGE_QUOTA);
+                      })));
+
+      try (ControllerClient parentControllerClient =
+          ControllerClient.constructClusterControllerClient(clusterName, parentControllerUrls)) {
+        // Enable VPJ to send liveness heartbeat.
+        props.put(BatchJobHeartbeatConfigs.HEARTBEAT_ENABLED_CONFIG.getConfigName(), true);
+        props.put(BatchJobHeartbeatConfigs.HEARTBEAT_STORE_NAME_CONFIG.getConfigName(), VPJ_HEARTBEAT_STORE_NAME);
+        // Prevent heartbeat from being deleted when the VPJ run finishes.
+        props.put(BatchJobHeartbeatConfigs.HEARTBEAT_LAST_HEARTBEAT_IS_DELETE_CONFIG.getConfigName(), false);
+
+        parentControllerClient.updateStore(VPJ_HEARTBEAT_STORE_NAME, new UpdateStoreQueryParams().setLeaderFollowerModel(true).setActiveActiveReplicationEnabled(true));
```

Comment:
Is it worthwhile to keep the test for heartbeat store in NativeReplication mode? If there's some value, probably we can use `True-or-Flase` provider for this.

Reply:
The reason we kept the test in native replication is originally heartbeat store is in native replication mode, now we plan to enable active/active for it, so I just move this test out.